src/h5web/guards.ts

```diff
@@ -87,19 +86,36 @@ export function isHardLink(link: HDF5Link): link is HDF5HardLink {
   return link.class === HDF5LinkClass.Hard;
 }
 
-export function hasSimpleShape<T extends HDF5Type>(
-  dataset: Dataset<HDF5Shape, T>
-): dataset is Dataset<HDF5SimpleShape, T> {
-  return dataset.shape.class === HDF5ShapeClass.Simple;
+export function isScalarShape(shape: Shape): shape is ScalarShape {
+  return shape !== null && shape.length === 0;
 }
 
 export function hasScalarShape<T extends HDF5Type>(
-  dataset: Dataset<HDF5Shape, T>
-): dataset is Dataset<HDF5ScalarShape, T> {
-  return dataset.shape.class === HDF5ShapeClass.Scalar;
+  dataset: Dataset<Shape, T>
+): dataset is Dataset<ScalarShape, T> {
+  return isScalarShape(dataset.shape);
+}
+
+export function hasSimpleShape<T extends HDF5Type>(
```

Comment:
`hasArrayShape` ? I feel `SimpleShape` is not very straightforward and we no longer need to match shape class.